Lib/asyncio/queues.py

```diff
@@ -170,15 +196,26 @@ async def get(self):
                     # the call.  Wake up the next in line.
                     self._wakeup_next(self._getters)
                 raise
-        return self.get_nowait()
+        # Don't run tests from self._get_nowait().
+        return self._get_and_wakeup_next()
 
     def get_nowait(self):
         """Remove and return an item from the queue.
 
-        Return an item if one is immediately available, else raise QueueEmpty.
+        Return an item if one is immediately available, else raise QueueEmpty
+        Return an item if there is no pending get task, else raise QueueEmptyWithPendingGetTasks
         """
         if self.empty():
             raise QueueEmpty
+        if self._getters:
+            raise QueueEmptyWithPendingGetTasks
```

Comment:
What if there are no getters yet but there's also no data? Does that also raise this exception? Or will it raise QueueEmpty?

Reply:
It will raise QueueEmpty. 
QueueEmptyWithPendingGetTasks will be raised only when queue and getter are not empty. 
The first waiter from getters should be the next one to get an item.